app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -52,26 +53,34 @@ def destroy
   end
 
   def deploy_all
-    @stages = Project.all.flat_map do |project|
-      environment = Environment.find(params[:environment_id]) if params[:environment_id]
+    environment = deploy_group.environment
+    deploys = deploy_group.stages.map do |stage|
+      template_stage = Stage.where(project: stage.project, environment: environment, is_template: true).first
 
-      stages = stages_in_same_environment(project, environment)
-      next unless deploy = stages.map(&:last_successful_deploy).compact.sort_by(&:created_at).last
-      stages.map { |s| [s, deploy] }
-    end.compact
-  end
+      last_success_deploy = template_stage.last_successful_deploy
+      next unless last_success_deploy
 
-  def deploy_all_now
-    deploys = params.require(:stages).map do |stage|
-      stage_id, reference = stage.split("-", 2)
-      stage = Stage.find(stage_id)
-      stage = new_stage_with_group(stage) unless only_to_current_group?(stage)
       deploy_service = DeployService.new(current_user)
-      deploy_service.deploy!(stage, reference: reference)
+      deploy_service.deploy!(stage, reference: last_success_deploy.reference)
     end
+
     redirect_to deploys_path(ids: deploys.map(&:id))
```

Comment:
I don't understand how this works. How does it redirect to a `show` page with multiple ids?